state/writer.go

```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+var res = regexp.MustCompile(`\${(.+)\.(.+)\.(.+)}`)
```

Comment:
Maybe it's also worth adding some docs/better name to it, so it's known what does it match.

Reply:
@kerak19 done